tests/integration/test_holo_hosting.js

```diff
@@ -0,0 +1,181 @@
+      const page_url = `${http_url}/html/index.html`;
+      const page = await create_page(page_url);
+      await page.setViewport({ width: 1442, height: 1341 });
+      
+      const pageTestUtils = new PageTestUtils(page);
+      pageTestUtils.logPageErrors();
+      pageTestUtils.describeJsHandleLogs();
+      
+      await page.exposeFunction('expect', (actualValue, expectedValue) => expect(actualValue).to.equal(expectedValue));
+
+      response = await page.evaluate(async function () {
+        console.log(" ----------------> 1");
+        const holoWebClient = new HoloWebSDK.Connection('http://localhost:24273', signal => console.log("Signal : ", signal),
+          {
+              logo_url: '../assets/holochain.png',
+              app_name: 'Test Holofuel App'
+          }
+        );
+        console.log(" ----------------> 2");
+
+	      holoWebClient.on("connected", console.log.bind(console));
+	      await holoWebClient.ready();
+        console.log(" ----------------> 3");
+
+	      window.holoWebClient = holoWebClient;
+        console.log(" ----------------> 4");
+
+	      await holoWebClient.signUp();
+        console.log(" ----------------> 5");
+        console.log("Called sign-up...");
+
+        const chaperone_modal = frames.find(f => f.url() === 'http://localhost:24273/?logo_url=http%3A%2F%2Flocalhost%3A8080%2Fassets%2Fholochain.png&app_name=My+Holofuel+App');
+        await page.waitForNavigation();
+
+        expect(!!chaperone_modal, true);
+
+        // visual test
+        const appLogo = await chaperone_modal.waitForSelector('img > .app-logo');
+        expect(appLogo.getAttribute("src"), "http://localhost:8080/assets/holochain.png");
+
+        const appName = await chaperone_modal.waitForSelector('h1 > span > .app-name');
+        expect(appName.value, "Test Holofuel App");
+
+        // input test -log in using chaperone modal
+        try {
+          const agentEmail = 'test@test.com'
+          const agentPassword = '123456'
+            
+          const emailInput = await chaperone_modal.waitForSelector('div > #sign-up #signup-email');
+          await chaperone_modal.click(emailInput);
+          await chaperone_modal.type(emailInput, agentEmail);
+          expect(emailInput, agentEmail);
+          
+          const passwordInput = await chaperone_modal.waitForSelector('#sign-up #signup-password');
+          await chaperone_modal.click(passwordInput);
+          await chaperone_modal.type(passwordInput, agentPassword);
+          expect(passwordInput, agentPassword);
+          
+          const passwordConfirmInput = await chaperone_modal.waitForSelector('#sign-up #signup-password-confirm');
+          await chaperone_modal.click(passwordConfirmInput);
+          await chaperone_modal.type(passwordConfirmInput, agentPassword);
+          expect(passwordConfirmInput, agentPassword);
+        } catch (error) {
+          console.log(typeof err.stack, err.stack.toString());
+          throw err
+        }
```

Comment:
Shouldn't this press the submit button?

Reply:
haha, yes, indeed it will. ;) I had prioritized the test setup and manual testing setup, so I haven't yet finished this integration test itself. :+1: